x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -602,87 +640,103 @@ export function buildTraceTree({
     hasInitializedChildren: false,
   };
 
-  return buildTree({ root, maxLevelOpen, waterfall, path });
+  const orphanRoots = waterfall.childrenByParentId[ROOT_ID]?.map((item) => ({
+    id: item.id,
+    item,
+    children: [],
+    level: 0,
+    expanded: isOpen,
+    childrenToLoad: 0,
+    hasInitializedChildren: false,
+  }));
+
+  const roots = [entryRoot, ...(orphanRoots ?? [])];
+
+  return buildTree({ roots, maxLevelOpen, waterfall, path });
 }
 
-export const convertTreeToList = (root: IWaterfallNode | null): IWaterfallNodeFlatten[] => {
-  if (!root) {
+export const convertTreeToList = (roots: IWaterfallNode[] | null): IWaterfallNodeFlatten[] => {
+  if (!roots) {
     return [];
   }
 
   const result: IWaterfallNodeFlatten[] = [];
-  const stack: IWaterfallNode[] = [root];
+  roots.forEach((root) => {
+    const stack: IWaterfallNode[] = [root];
```

Comment:
Same here - maybe the easiest is to that to: 

```suggestion
const stack: IWaterfallNode[] = [...roots];
```

wdyt?